frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java

```diff
@@ -42,6 +47,19 @@ public GitMachetePanel(Project project) {
     val selectionComponent = selectedGitRepositoryProvider.getSelectionComponent();
     val graphTable = getGraphTable();
 
+    project.getMessageBus().connect().subscribe(VirtualFileManager.VFS_CHANGES, new BulkFileListener() {
+      @Override
+      public void after(java.util.List<? extends VFileEvent> events) {
+        for (val event : events) {
+          if (event instanceof VFileContentChangeEvent) {
+            if (MacheteFileType.instance.equals(((VFileContentChangeEvent) event).getFile().getFileType())) {
+              graphTable.queueRepositoryUpdateAndModelRefresh();
```

Comment:
Does it make sense to skip the update if the tab is NOT opened yet?

I'm asking esp. wrt. the problem of premature autodiscover... although a change to .git/machete will bump the file's modification date, so autodiscover shouldn't occur anyway :trollface: